src/rgw/rgw_admin_argument_parsing.h

```diff
@@ -0,0 +1,98 @@
+                                       std::list<std::string>& tags_add, std::list<std::string>& tags_rm,
+                                       std::string& tier_type, bool& tier_type_specified,
+                                       map<std::string, std::string, ltstr_nocase>& tier_config_add,
+                                       map<std::string, std::string, ltstr_nocase>& tier_config_rm,
+                                       std::string& zone_id, std::string& zone_name,
+                                       std::string& zone_new_name, std::string& zonegroup_id,
+                                       std::string& zonegroup_name, std::string& zonegroup_new_name);
+
+int parse_quota_commandline_params(std::vector<const char*>& args, std::string& quota_scope,
+                                   int64_t& max_size, bool& have_max_size,
+                                   int64_t& max_objects, bool& have_max_objects);
+
+int parse_role_commandline_params(std::vector<const char*>& args, std::string& role_name,
+                                  std::string& policy_name, std::string& assume_role_doc,
+                                  std::string& perm_policy_doc, std::string& path,
+                                  std::string& path_prefix);
+
+int parse_user_commandline_params(std::vector<const char*>& args,
+                                  std::string& subuser, std::string& display_name,
+                                  std::string& user_email, RGWUserAdminOpState& user_op,
+                                  std::string& caps, std::string& op_mask_str, int& key_type,
+                                  int& purge_keys, int& max_buckets, bool& max_buckets_specified,
+                                  int& admin, bool& admin_specified, int& system, bool& system_specified,
+                                  bool& set_temp_url_key, map<int, std::string>& temp_url_keys,
+                                  std::string& access, uint32_t& perm_mask, bool& set_perm,
+                                  int& sync_stats);
+
+int parse_other_commandline_params(std::vector<const char*>& args, std::string& pool_name,
+                                   rgw_pool& pool, std::string& daemon_id, bool& specified_daemon_id,
+                                   std::string& replica_log_type_str, ReplicaLogType& replica_log_type,
+                                   int& extra_info, int& include_all, std::string& client_id,
+                                   std::string& op_id, std::string& state_str, int& show_log_entries,
+                                   int& show_log_sum, int& skip_zero_entries,
+                                   std::map<std::string, bool>& categories,
+                                   uint64_t& orphan_stale_secs, std::string& job_id);
+
+class RgwAdminCommandGroupHandlerFactory {
+public:
+  static RgwAdminCommandGroupHandler* get_command_group_handler(std::vector<const char*>& args);
```

Comment:
why is every thing static?

Reply:
Mapping from string to the corresponding command group is the same for every instance of this class, so is static. Additionally, I don't see any state this class should hold, so I feel that static methods work just fine.